configure.ac

```diff
@@ -238,6 +238,31 @@ AC_TRY_LINK([#include <stdint.h>],
     ],
     [AC_MSG_RESULT(no)])
 
+dnl Check for gcc memory model aware built-in atomics
+dnl If supported check to see if not internal to compiler
+AC_MSG_CHECKING(compiler support for built-in memory model aware atomics)
+AC_TRY_LINK([#include <stdint.h>],
+    [uint64_t d;
+     uint64_t s;
+     uint64_t c;
+     uint64_t r;
+      r = __atomic_fetch_add(&d, s, __ATOMIC_SEQ_CST);
+      __atomic_load(&d, &r, __ATOMIC_SEQ_CST);
+      __atomic_exchange(&d, &s, &r, __ATOMIC_SEQ_CST);
+      __atomic_compare_exchange(&d,&c,&s,0, __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST);
+     #if defined(__PPC__) && !defined(__PPC64__)
+       #error compiler built-in memory model aware atomics are not supported on PowerPC 32-bit
+     #else
+       return 0;
+     #endif
+    ],
+    [
+        AC_MSG_RESULT(yes)
+        AC_SEARCH_LIBS([__atomic_load_8], [atomic])
```

Comment:
The travis build was complaining about an undefined symbol (__atomic_load_8). It appears that for some gcc versions and/or architectures even though they support the built-in they require an external library be linked. @jswaro, @jsquyres is this a suitable way to handle this?

Reply:
Seems reasonable to me. @jsquyres would probably know a really elegant solution for something like this.

Reply:
Check out [`AC_SEARCH_LIBS`](https://www.gnu.org/software/autoconf/manual/autoconf-2.66/html_node/Libraries.html), it may save you a bunch of this extra m4 code...?